RAG-LLM GitOps Pattern on MS Azure

abhatt-rh opened the pull request:
Documentation for RAG-LLM Pattern on MS Azure

Issues: https://issues.redhat.com/browse/TELCODOCS-2392

dminnear-rh commented on content/patterns/azure-rag-llm-gitops/az-ragllm-getting-started.adoc:
```diff
@@ -0,0 +1,149 @@
+. Record the generated *Server name*, *Server admin login*, and *Password*. These credentials will be used later when creating secrets.
+. On the *Networking* tab, ensure that *Allow Azure services and resources to access this server is set* to *Yes*. This allows your ARO cluster to connect to the database.
+. Click *Review + create*, and then click *Create*.
+
+Wait for the SQL Server deployment to complete and become active before proceeding.
+
+[id="creating-secret-credentials_{context}"]
+=== Creating secret credentials
+
+To securely store your sensitive credentials, create a YAML file named `~/values-secret-rag-llm-gitops.yaml`. This file is used during the pattern deployment; however, you must not commit it to your Git repository.
+
+[IMPORTANT]
+====
+If you’re not using Azure SQL Server, omit the entire `azuresql` section.
+====
+
+[source,yaml]
+----
+# ~/values-secret-rag-llm-gitops.yaml
+# Replace placeholders with your actual credentials
+version: "2.0"
+
+secrets:
+  - name: hfmodel
+    fields:
+      - name: hftoken <1>
+        value: <hf_your_huggingface_token>
+  - name: azuresql
+    fields:
+      - name: user <2>
+        value: <adminuser>
+      - name: password <3>
+        value: <your_password>
+      - name: server <4>
+        value: <yourservername.database.windows.net>
+----
+<1> Specify your Hugging Face token.
+<2> Specify the username for your Azure SQL server.
+<3> Specify the password for your Azure SQL server.
+<4> Specify the fully qualified of your Azure SQL server name.
```
Since we use the local deploy of MS SQL server by default, we can omit the section about creating the Azure SQL Server instance. We just need to make sure to update the secrets to include the password for SQL server. (That said, if you omit the password, one will be generated automatically. The only secret you absolutely must include is the HF token.)

```suggestion
=== Creating secret credentials

To securely store your sensitive credentials, create a YAML file named `~/values-secret-rag-llm-gitops.yaml`. This file is used during the pattern deployment; however, you must not commit it to your Git repository.

[source,yaml]
----
# ~/values-secret-rag-llm-gitops.yaml
# Replace placeholders with your actual credentials
version: "2.0"

secrets:
  - name: hfmodel
    fields:
      - name: hftoken <1>
        value: <hf_your_huggingface_token>
  - name: mssql
    fields:
    - name: sa-pass <2>
      value: <password_for_sa_user>
----
<1> Specify your Hugging Face token.
<2> Specify the password to give the SA user on the MS SQL Server instance.
```

dminnear-rh commented on content/patterns/azure-rag-llm-gitops/az-ragllm-getting-started.adoc:
```diff
@@ -0,0 +1,149 @@
+To install the RAG-LLM GitOps Pattern on Microsoft Azure, you must complete the following setup and configurations:
+
+* xref:creating-huggingface-token[Create a Hugging face token]
+* xref:deploying-azure-sql-server[Deploy Azure SQL]
```
We can remove this section

abhatt-rh commented:
@dminnear-rh Thank you again for your prompt review. I have addressed your comments. IF all looks OK, pls approve the PR and feel free to merge it or tag me for further changes.
